core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -1423,48 +1425,50 @@ sealed trait ZIO[-R, +E, +A]
     leftScope: FiberScope = null,
     rightScope: FiberScope = null
   )(implicit trace: Trace): ZIO[R1, E2, C] =
-    ZIO.withFiberRuntime[R1, E2, C] { (parentFiber, parentStatus) =>
-      import java.util.concurrent.atomic.AtomicBoolean
-
-      implicit val unsafe: Unsafe = Unsafe
-
-      val parentRuntimeFlags = parentStatus.runtimeFlags
-
-      @inline def complete[E0, E1, A, B](
-        winner: Fiber.Runtime[E0, A],
-        loser: Fiber.Runtime[E1, B],
-        cont: (Fiber.Runtime[E0, A], Fiber.Runtime[E1, B]) => ZIO[R1, E2, C],
-        ab: AtomicBoolean,
-        cb: ZIO[R1, E2, C] => Any
-      ): Any =
-        if (ab.compareAndSet(true, false)) {
-          cb(cont(winner, loser))
-        }
+    ZIO.transplant { graft =>
```

Comment:
Note that you could potentially add an apply method in `Grafter`s companion object that accepts a `Fiber.Runtime` method to avoid duplicating the logic